plugins/outputs/otlp/otlp.go

```diff
@@ -0,0 +1,379 @@
+						}
+					}
+				case int64:
+					point.Data = &otlpmetricpb.Metric_IntSum{
+						IntSum: monotonicIntegerPoint(labels, ts, currentTs, v),
+					}
+				case float64:
+					point.Data = &otlpmetricpb.Metric_DoubleSum{
+						DoubleSum: monotonicDoublePoint(labels, ts, currentTs, v),
+					}
+				case bool:
+					if v {
+						point.Data = &otlpmetricpb.Metric_IntSum{
+							IntSum: monotonicIntegerPoint(labels, ts, currentTs, 1),
+						}
+					} else {
+						point.Data = &otlpmetricpb.Metric_IntSum{
+							IntSum: monotonicIntegerPoint(labels, ts, currentTs, 0),
+						}
+					}
+				case string:
+					o.Log.Error("get type failed: unsupported telegraf value type string")
+					continue
+				default:
+					o.Log.Errorf("get type failed: unsupported telegraf value type %v\n", f.Value)
+					continue
+				}
+			case telegraf.Gauge, telegraf.Untyped:
+				switch v := f.Value.(type) {
+				case uint64:
+					if v <= uint64(MaxInt) {
+						point.Data = &otlpmetricpb.Metric_IntGauge{
+							IntGauge: intGauge(labels, ts, int64(v)),
+						}
+					} else {
+						point.Data = &otlpmetricpb.Metric_IntGauge{
+							IntGauge: intGauge(labels, ts, int64(MaxInt)),
+						}
+					}
+				case int64:
+					point.Data = &otlpmetricpb.Metric_IntGauge{
+						IntGauge: intGauge(labels, ts, v),
+					}
+				case float64:
+					point.Data = &otlpmetricpb.Metric_DoubleGauge{
+						DoubleGauge: doubleGauge(labels, ts, v),
+					}
+				case bool:
+					if v {
+						point.Data = &otlpmetricpb.Metric_IntGauge{
+							IntGauge: intGauge(labels, ts, 1),
+						}
+					} else {
+						point.Data = &otlpmetricpb.Metric_IntGauge{
+							IntGauge: intGauge(labels, ts, 0),
+						}
+					}
+				case string:
+					o.Log.Error("get type failed: unsupported telegraf value type string")
+					continue
+				default:
+					o.Log.Errorf("get type failed: unsupported telegraf value type %v\n", f.Value)
+					continue
+				}
+			// TODO: add support for histogram & summary
+			case telegraf.Histogram, telegraf.Summary:
+				fallthrough
+			default:
+				o.Log.Errorf("get type failed: unsupported telegraf metric kind %v\n", m.Type())
+				continue
+			}
+			samples = append(samples, sample)
+		}
+	}
+
+	if err := o.client.Store(&metricsService.ExportMetricsServiceRequest{
+		ResourceMetrics: samples,
+	}); err != nil {
+		o.Log.Errorf("unable to write to endpoint: %s", err)
+		return err
+	}
+	return nil
+}
+
+// Close will terminate the session to the backend, returning error if an issue arises.
+func (o *OTLP) Close() error {
+	return o.client.Close()
+}
+
+// SampleConfig returns the formatted sample configuration for the plugin.
+func (o *OTLP) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns the human-readable function definition of the plugin.
+func (o *OTLP) Description() string {
+	return "Configuration for OTLP to send metrics to"
+}
+
+func newOTLP() *OTLP {
+	return &OTLP{}
+}
+
```

Comment:
Please remove this function and initialize the structure directly.

Reply:
Done